test-cypress/integration/cluster/cluster-configuration.spec.js

```diff
@@ -0,0 +1,277 @@
+                    local: false
+                }
+            ];
+
+            nodeData.forEach((data, index) => {
+                ClusterConfigurationSteps.getNodeLink(index)
+                    .should('have.attr', 'href', data.url)
+                    .and('contain.text', data.url);
+
+                ClusterConfigurationSteps.getNodeRPCAddress(index)
+                    .should('contain.text', data.rpcAddress);
+
+                ClusterConfigurationSteps.getNodeState(index)
+                    .should('contain.text', data.state);
+
+                if (data.local) {
+                    ClusterConfigurationSteps.isNodeLocal(index).should('exist');
+                } else {
+                    ClusterConfigurationSteps.isNodeLocal(index).should('not.exist');
+                }
+            });
+        });
+    });
+
+    context('Multi-region', () => {
+        beforeEach(() => {
+            // Given there is an existing cluster created
+            ClusterStubs.stubClusterConfig();
+            ClusterStubs.stubClusterGroupStatus();
+            ClusterStubs.stubClusterNodeStatus();
+            RemoteLocationStubs.stubRemoteLocationFilter();
+            RemoteLocationStubs.stubRemoteLocationStatusInCluster();
+        });
+
+        context('Primary cluster', () => {
+            it('should be able to add and delete tags', () => {
+                const tagName = 'test';
+                // Given I have opened the cluster management page
+                ClusterPageSteps.visit();
+                // When I click on edit properties and open Nodes tab
+                ClusterPageSteps.previewClusterConfig();
+                ClusterConfigurationSteps.selectTab(2);
+                // I expect to see
+                ClusterConfigurationSteps.getMultiRegionHeader().should('contain.text', 'Primary cluster');
+                ClusterConfigurationSteps.getAddTagButton().should('be.visible').and('not.be.disabled');
+                ClusterConfigurationSteps.getEnableSecondaryModeButton().should('be.visible').and('not.be.disabled');
+                ClusterConfigurationSteps.getTagsTable().should('not.exist');
+
+                ClusterStubs.stubAddTag(tagName);
+                ClusterConfigurationSteps.clickAddTagButton();
+                ClusterConfigurationSteps.typeTag(tagName);
+                ClusterConfigurationSteps.clickSubmitTagButton();
+                cy.wait('@add-tag').then((interception) => {
+                    expect(interception.request.body).to.deep.equal({tag: tagName});
+                });
+
+                ClusterStubs.stubClusterGroupStatusWithTag();
+                cy.wait('@3-nodes-cluster-group-status-tag').then(() => {
+                    // Assert the tags table contains the expected tag
+                    ClusterConfigurationSteps.getTagsTable().should('be.visible');
+                    ClusterConfigurationSteps.getTagsTableRows().should('contain.text', tagName);
```

Comment:
Move these out and remove the then callback. You don't need it, because cypress automatically waits for the request to happen and then proceeds with the next steps. You only need to handle the promise if you have to deal with the interception/request itself.

Reply:
done